buildpack/runtime_components/storage.py

```diff
@@ -65,17 +76,55 @@ def _get_s3_specific_config(vcap_services, m2ee):
     v2_auth = os.getenv("S3_USE_V2_AUTH", v2_auth).lower() == "true"
     sse = os.getenv("S3_USE_SSE", "").lower() == "true"
 
-    if not (access_key and secret and bucket):
+    if (access_key and secret and bucket):
+        logging.info("S3 config detected, activating external file store")
+        config = {
+            "com.mendix.core.StorageService": "com.mendix.storage.s3",
+            "com.mendix.storage.s3.AccessKeyId": access_key,
+            "com.mendix.storage.s3.SecretAccessKey": secret,
+            "com.mendix.storage.s3.BucketName": bucket,
+        }
+    elif (tvm_endpoint and tvm_username and tvm_password and bucket):
+        if m2ee.config.get_runtime_version() < 9.2:
+            logging.info("S3 config detected, fetching IAM credentials from TVM")
+            response = requests.get(
```

Comment:
I would use what is described here: https://docs.python-requests.org/en/master/user/authentication/
```
requests.get('https://api.github.com/user', auth=('user', 'pass'))
```
Then `_get_credential()` is not required.